src/home/templates/partials/_table.html

```diff
@@ -1,420 +1,580 @@
+        });
+        
+
+        // Pagination controls
+        const paginationDiv = document.createElement("div");
+        paginationDiv.classList.add("flex", "justify-between", "mt-4");
+
+        if (has_previous) {
+          const prevButton = document.createElement("button");
+          prevButton.classList.add(
+            "bg-blue-500",
+            "text-white",
+            "px-4",
+            "py-2",
+            "rounded"
+          );
+          prevButton.textContent = "Previous";
+          prevButton.addEventListener("click", () =>
+            fetchAndDisplayReviews(serviceId, current_page - 1)
+          );
+          paginationDiv.appendChild(prevButton);
+        }
 
+        if (has_next) {
+          const nextButton = document.createElement("button");
+          nextButton.classList.add(
+            "bg-blue-500",
+            "text-white",
+            "px-4",
+            "py-2",
+            "rounded"
+          );
+          nextButton.textContent = "Next";
+          nextButton.addEventListener("click", () =>
+            fetchAndDisplayReviews(serviceId, current_page + 1)
+          );
+          paginationDiv.appendChild(nextButton);
+        }
 
-            const descriptionElement = document.getElementById('serviceDescription');
-            descriptionElement.innerHTML = '';  // Clear previous content
-
-            const heading = document.createElement('h3');
-            heading.textContent = 'Additional Descriptive Details:';
-            heading.style.marginBottom = '10px';
-            heading.style.fontSize = '1.1em';
-            heading.style.fontWeight = 'bold';
-            descriptionElement.appendChild(heading);
-
-            // Check if Description exists and is an object
-            if (service.Description && typeof service.Description === 'object') {
-                let hasDescription = false;
-                const table = document.createElement('table');
-
-                for (const [key, value] of Object.entries(service.Description)) {
-                    if (value !== null && value !== '') {
-                        hasDescription = true;
-                        const tr = document.createElement('tr');
+        reviewsContainer.appendChild(paginationDiv);
+      } catch (error) {
+        console.error("Error fetching reviews:", error);
+        alert("Failed to load reviews. Please try again.");
+      }
+    }
 
-                        const th = document.createElement('th');
-                        th.textContent = `${key.replace(/_/g, ' ')}:`;
+    // Expose the function globally
+    window.fetchAndDisplayReviews = fetchAndDisplayReviews;
 
-                        const td = document.createElement('td');
-                        td.innerHTML = value.replace(/\n/g, '<br>');
+    async function handleEditReview(review) {
+      try {
+        const newRating = prompt(
+          "Enter a new rating (1-5):",
+          review.RatingStars
+        );
```

Comment:
instead of prompts can we use a modal instead? i think that would be better UX. would also be inline with how we allow edit operations for comments and forum posts